modules/core/src/main/java/org/apache/ignite/internal/commandline/checkpointing/CheckpointingForceTask.java

```diff
@@ -0,0 +1,136 @@
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Task for checkpointing operation
+ */
+@GridInternal
+public class CheckpointingForceTask extends VisorMultiNodeTask<VoidDto, CheckpointingForceResult, NodeCheckpointingResult> {
+
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    @Override protected VisorJob<VoidDto, NodeCheckpointingResult> job(VoidDto arg) {
+        return new CheckpointingForceJob(debug);
+    }
+
+    @Nullable
+    @Override protected CheckpointingForceResult reduce0(List<ComputeJobResult> results) throws IgniteException {
+        CheckpointingForceResult res = new CheckpointingForceResult();
+
+        for (ComputeJobResult result : results) {
+            NodeCheckpointingResult nodeRes = result.getData();
+            res.status().put(nodeRes.nodeId(), nodeRes);
+        }
+        return res;
+    }
+
+    /** */
+    private static class CheckpointingForceJob extends VisorJob<VoidDto, NodeCheckpointingResult> {
+
+        /** */
+        private static final long serialVersionUID = 0L;
+
+        /** */
+        @JobContextResource
+        private ComputeJobContext jobCtx;
+
+        /** */
+        private GridFutureAdapter fut;
+
+        private NodeCheckpointingResult res;
+
+        long startTime;
+
+        /**
+         * @param arg Argument.
+         * @param debug Debug.
+         */
+        public CheckpointingForceJob(boolean debug) {
+            super(null, debug);
+        }
+
+        /** {@inheritDoc} */
+        @Override protected NodeCheckpointingResult run(@Nullable VoidDto arg) throws IgniteException {
+            if (fut == null) {
+                res = new NodeCheckpointingResult();
+                res.nodeId(ignite.localNode().id());
+
+                startTime = U.currentTimeMillis();
+                CheckpointProgress progress = ignite.context().cache().context().database().forceCheckpoint("force");
+                if (progress == null) {
+                    res.durationMillis(U.currentTimeMillis() - startTime);
+                    return res;
+                }
+
+                fut = progress.futureFor(CheckpointState.FINISHED);
+                if (fut.isDone())
+                    getFuture(fut, res);
+
+                else {
+                    jobCtx.holdcc();
+
+                    fut.listen(new IgniteInClosure<IgniteInternalFuture>() {
+                        @Override public void apply(IgniteInternalFuture f) {
+                            res.durationMillis(U.currentTimeMillis() - startTime);
+
+                            if (f.isCancelled())
+                                res.error(f.error());
+
+                            jobCtx.callcc();
+                        }
+                    });
+                }
+
+                return null;
+            }
+            getFuture(fut, res);
+
+            return res;
+        }
+    }
+
+    private static Object getFuture(GridFutureAdapter fut, NodeCheckpointingResult res) {
```

Comment:
The name of this method is not clear to me. Its name does not correlate with what this method actually does.